src/json/any.cr

```diff
@@ -120,14 +120,17 @@ struct JSON::Any
   # Traverses the depth of a structure and returns the value.
   # Returns `nil` if not found.
   def dig?(key : String | Int, *subkeys)
-    if (value = self[key]?) && value.responds_to?(:dig?)
+    if value = self[key]?
       value.dig?(*subkeys)
     end
```

Comment:
Just a nitpick, totally optional, but I think this can be written like:

```crystal
self[key]?.try &.dig?(*subkeys)
```

Same thing in `YAML::Any`.

Reply:
https://github.com/crystal-lang/crystal/pull/8745#discussion_r374990371